### PR TITLE
Upgrade golangci-lint to v1.60.1

### PR DESCRIPTION
**Pull request by aknuds1**
I propose upgrading golangci-lint from v1.59.1 to v1.60.1. What triggered me to make the upgrade is that linting Prometheus with v1.59.1 on Linux causes a massive memory leak (my 64 GB RAM desktop runs out of memory). The memory leak doesn't occur with v1.60.1, but it finds a lot of issues that I also fix in this PR.

### CODE REVIEW

**Review by dgl — APPROVED** [review]
Sounds like possibly this is related to some mismatches of Go versions (there's some discussion in https://github.com/golangci/golangci-lint/issues/4909, although I don't see this using 1.23 yet).

File: notifier/notifier.go [hunk 1]
```diff
@@ -674,7 +674,7 @@ func (n *Manager) sendOne(ctx context.Context, c *http.Client, url string, b []b
 	}()
 
 	// Any HTTP status 2xx is OK.
-	//nolint:usestdlibvars
+
```
[the remark this answers is not in the document]
Comment (aknuds1) [reply notifier/notifier.go]: Automated lint fixing. Missed this empty line, sorry. Fixed now.

**Comment by aknuds1** [comment]
> Sounds like possibly this is related to some mismatches of Go versions

@dgl Yes, it could be because I'm on Go 1.23, although Prometheus' `toolchain` is still on v1.22, while the issue connects the problem to the `toolchain`.

**Review by ArthurSens — APPROVED** [review]

File: template/template.go [hunk 2]
```diff
@@ -166,7 +166,7 @@ func NewTemplateExpander(
 				return html_template.HTML(text)
 			},
 			"match":     regexp.MatchString,
-			"title":     strings.Title, //nolint:staticcheck
+			"title":     strings.Title,
```
Comment (bboreham) [inline template/template.go:169]: This one is weird because `strings.Title` is deprecated; it should get a warning.

Comment (aknuds1) [reply template/template.go:169]: @bboreham Guessing this was a bug in v1.60.1, since this is caught again in v1.60.2.

**Pull request merged**